src/main/java/io/parsingdata/util/Visualizer.java

```diff
@@ -49,7 +51,40 @@ public Visualizer() {
 
             @Override
             public String toString(final ParseValue value) {
-                return String.format("\"[0x%s]: %s\"", Long.toHexString(value.offset).toUpperCase(), value.asNumeric());
+                // Use value.getBytes() as String, max length 20
+                byte[] bytes = value.getValue();
+                if (bytes.length > MAX_STRING_SIZE) {
+                    bytes = new byte[MAX_STRING_SIZE];
+                    System.arraycopy(value.getValue(), 0, bytes, 0, bytes.length);
+                }
+
+                // Check if all bytes are a character in UTF-8
+                boolean isString = true;
+                for (final byte character : bytes) {
+                    final int charValue = character & 0xff;
+                    if (charValue < ' ' || charValue > '~') {
+                        isString = false;
+                        break;
+                    }
+                }
+
+                final StringBuilder builder = new StringBuilder();
+                if (value.getValue().length == 4 || value.getValue().length == 8) {
+                    // Possible Integer or Long
+                    builder.append("0x");
+                    builder.append(Long.toHexString(value.asNumeric().longValue()).toUpperCase());
+                    builder.append(' ');
+                    builder.append(value.asNumeric().longValue());
+                    builder.append('L');
+                    builder.append(isString ? " " : "");
+                }
+                if (isString) {
+                    // Possible String
+                    builder.append(new String(bytes, StandardCharsets.UTF_8));
```

Comment:
Would the character encoding of the ParseValue itself be of any use in order to support different encoding formats?